Ability to set password in Open Distro Elasticsearch, Filebeat and Kibana (#1994)
epiphany-platform/epiphany: docs/home/howto/MONITORING.md

By default Kibana adjusts the UTC time in `@timestamp` to the browser's local timezone. This can be changed in `Management` > `Advanced Settings` > `Timezone for date formatting`.

> ## How to configure default user password to Kibana - Open Distro and Open Distro for Elasticsearch and Filebeat

Reviewer: ```suggestion
## How to configure default user passwords for Kibana - Open Distro, Open Distro for Elasticsearch and Filebeat
```

To configure admin password for Kibana - Open Distro and Open Distro for Elasticsearch you need to follow the procedure below.
There are separated procedures for Open Distro and for Logging role as most of the times in clean Open Distro installation users:

> kibanaserver and logstash are not required to be present.

Reviewer: ```suggestion
There are separate procedures for `opendistro-for-elasticsearch` role and `logging` role since most of the times for `opendistro-for-elasticsearch`, `kibanaserver` and `logstash` users are not required to be present.
```

### Open Distro for Elasticsearch

By default Epiphany removes all demo users from except admin user in Open Distro role. Those users are listed in section 
demo_users_to_remove in configuration/opendistro-for-elasticsearch. If you want to leave kibanaserver user (needed by default 
Epiphany installation of Kibana) or logstash (needed by default Epiphany installation of Filebeat) you need to remove each specific
user from demo_users_to_remove list and to perform configuration by Epiphany kibanaserver_user_active to true for kibanaserver user or
logstash_user_active for logstash user. We strongly advice to set different passwords for admin and kibanaserver or logstash user. To 
change admin password please change value under admin_password key, for kibanaserver and logstash change respectively values under keys

> kibanaserver_password and logstash_password.

Reviewer: ```suggestion
By default Epiphany removes all demo users except `admin` user. Those users are listed in
`demo_users_to_remove` section of `configuration/opendistro-for-elasticsearch` doc. If you want to keep `kibanaserver` user (needed by default 
Epiphany installation of Kibana) you need to remove it from `demo_users_to_remove` list and set `kibanaserver_user_active` to `true` in order to change the default password. We strongly advice to set different password for each user. To 
change `admin` user's password change value for `admin_password` key. For `kibanaserver` and `logstash` change values for 
`kibanaserver_password` and `logstash_password` keys respectively.
```

```yaml
kind: configuration/opendistro-for-elasticsearch
title: Open Distro for Elasticsearch Config
name: default
specification:
  ...
  admin_password: YOUR_PASSWORD
  kibanaserver_password: YOUR_PASSWORD
  kibanaserver_user_active: false
  logstash_password: YOUR_PASSWORD
  logstash_user_active: false
  demo_users_to_remove:
  - kibanaro
  - readall
  - snapshotrestore
  - logstash
  - kibanaserver
```

### Logging role

#### - Logging role

By default Epiphany removes users that are listed in section demo_users_to_remove in configuration/opendistro-for-elasticsearch. By default
kibanaserver user (needed by default Epiphany installation of Kibana) and logstash (needed by default Epiphany installation of
Filebeat) are not removed. If you want to perform configuration by Epiphany kibanaserver_user_active to true for kibanaserver user or
logstash_user_active for logstash user. By default in logging role those settings are already set to true. We strongly advice to set
different passwords for admin and kibanaserver and logstash user. To change admin password please change value under admin_password key, for

> kibanaserver and logstash change respectively values under keys kibanaserver_password and logstash_password. Please remember to also change passwords accordingly in configuration of Filebeat and Kibana.

Reviewer: ```suggestion
By default Epiphany removes users that are listed in `demo_users_to_remove` section of 
`configuration/opendistro-for-elasticsearch` doc. By default, `kibanaserver` user (needed by default Epiphany installation of Kibana) and `logstash` (needed by default Epiphany installation of
Filebeat) are not removed. If you want to perform configuration by Epiphany, set `kibanaserver_user_active` to `true` for `kibanaserver` user or
`logstash_user_active` for `logstash` user. For `logging` role, those settings are already set to `true` by default. We strongly advice to set different password for each user. To change `admin` user's password change value for `admin_password` key. For `kibanaserver` and `logstash` change values for `kibanaserver_password` and `logstash_password` keys respectively. Please remember to also change passwords accordingly in configuration of Filebeat and Kibana.
```

> kibanaserver and logstash change respectively values under keys kibanaserver_password and logstash_password. Please remember to also change passwords accordingly in configuration of Filebeat and Kibana.

Reviewer: > Please remember to also change passwords accordingly in configuration of Filebeat and Kibana.

Why we have this sentence? Aren't these passwords set by Epiphany automatically (Jinja templates)?

Author: They are, but you need to separately set them also in other roles.

```yaml
kind: configuration/logging
title: Logging Config
name: default
specification:
  ...
  admin_password: YOUR_PASSWORD
  kibanaserver_password: YOUR_PASSWORD
  kibanaserver_user_active: true
  logstash_password: YOUR_PASSWORD
  logstash_user_active: true
  demo_users_to_remove:
  - kibanaro
  - readall
  - snapshotrestore
```

#### - Kibana role

To set password of kibanaserver user that is used by Kibana for communication with Open Distro Elasticsearch backend please set value under 
key kibanaserver_password in configuration/kibana like in the example specified below. For this moment there is no possibility to specify

> a different user name than kibanaserver for Kibana and Open Distro Elasticsearch backend.

Reviewer: ```suggestion
To set password of `kibanaserver` user, which is used by Kibana for communication with Open Distro Elasticsearch backend, change value for `kibanaserver_password` key in `configuration/kibana` doc like in the example below. For this moment there is no possibility to specify
a different user name than `kibanaserver` for Kibana and Open Distro Elasticsearch backend.
```

```yaml
kind: configuration/kibana
title: "Kibana"
name: default
specification:
  ...
  kibanaserver_password: YOUR_PASSWORD
```

#### - Filebeat role

To set password of logstash user that is used by Filebeat for communication with Open Distro Elasticsearch backend please set value under
key logstash_password in configuration/kibana like in the example specified below. For this moment there is no possibility to specify

> a different user name than logstash for Filebeat and Open Distro Elasticsearch backend.

Reviewer: ```suggestion
To set password of `logstash` user, which is used by Filebeat for communication with Open Distro Elasticsearch backend, change value for `logstash_password` key in `configuration/filebeat` doc like in the example below. For this moment there is no possibility to specify
a different user name than `logstash` for Filebeat and Open Distro Elasticsearch backend.
```

```yaml
kind: configuration/filebeat
title: Filebeat
name: default
specification:
  ...
  logstash_password: YOUR_PASSWORD
```

### Upgrade of Elasticsearch, Kibana and Filebeat

> During upgrade Epiphany is taking kibanaserver (for Kibana) and logstash user (for Filebeat) passwords and will reapply them to upgraded configuration of Filebeat and Kibana. Epiphany ugrade of Open Distro, Kibana and Filebeat will fail if kibanaserver and logstash usernames were changed in configuration of Kibana and Filebeat and on Open Distro for Elasticsearch.

Reviewer 2: Typo - 'ugrade'

Author: Fixed.

> During upgrade Epiphany is taking kibanaserver (for Kibana) and logstash user (for Filebeat) passwords and will reapply them to upgraded configuration of Filebeat and Kibana. Epiphany upgrade of Open Distro, Kibana and Filebeat will fail if kibanaserver and logstash usernames were changed in configuration of Kibana and Filebeat and on Open Distro for Elasticsearch.

Reviewer: ```suggestion
During upgrade Epiphany takes `kibanaserver` (for Kibana) and `logstash` (for Filebeat) user passwords and re-applies them to upgraded configuration of Filebeat and Kibana. Epiphany upgrade of Open Distro, Kibana or Filebeat will fail if `kibanaserver` or `logstash` usernames were changed in configuration of Kibana, Filebeat or Open Distro for Elasticsearch.
```
